controllers/nodehealthcheck_controller.go

```diff
@@ -341,7 +332,21 @@ func (r *NodeHealthCheckReconciler) remediate(node *v1.Node, nhc *remediationv1a
 		}
 	}
 
-	// set control plane marker label
+	currentTemplate, timeout, err := rm.GetCurrentTemplateWithTimeout(node, nhc)
+	if err != nil {
+		if _, ok := err.(resources.NoTemplateLeftError); ok {
+			log.Error(err, "Remediation timed out, and no template left to try")
```

Comment:
Nit: I'm not sure an error is justified in this case, IMO info is fine

Reply:
This means that remediation failed, IMHO a good reason to log it as error?

Reply:
I'm not so sure .
ATM I don't think NHC logs an error in case a remediation is failing.
One argument against it could be that NHC responsibility is the detection of error and notification of those error to the remediators and not the actual remediation, so such error would be more appropriate to be logged within the remediators themselves.
It's certainly not something that I feel  strongly about so feel free to leave unchanged.

Reply:
Yes, NHC isn't responsible for a single issue with one remediator. But the remediators a) don't know about the timeout and b) don't know that there might be other remediators involved. So NHC is the only component which knows the overall state of remediation. So I prefer to keep the log.